File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/LatLngStr.java
package closer.vlllage.com.closer.handler.helpers;

import com.google.android.gms.maps.model.LatLng;

import java.util.List;

import closer.vlllage.com.closer.pool.PoolMember;

public class LatLngStr extends PoolMember {

    public String from(LatLng latLng) {
        return latLng.latitude + "," + latLng.longitude;
    }

    public LatLng to(String latLng) {
        String[] parts = latLng.split(",");
        return new LatLng(Double.valueOf(parts[0]), Double.valueOf(parts[1]));
    }

    public LatLng to(List<Double> latLng) {
        if (latLng == null) {
            return null;
        }

        return new LatLng(latLng.get(0), latLng.get(1));
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/LatLngStr.kt
package closer.vlllage.com.closer.handler.helpers

import at.bluesource.choicesdk.maps.common.LatLng
import com.queatz.on.On

class LatLngStr constructor(private val on: On) {

    fun from(latLng: LatLng): String {
        return latLng.latitude.toString() + "," + latLng.longitude
    }

    fun to(latLng: String): LatLng {
        val parts = latLng.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return LatLng(java.lang.Double.valueOf(parts[0]), java.lang.Double.valueOf(parts[1]))
    }

    fun to(latLng: List<Double>?): LatLng? {
        return if (latLng == null) {
            null
        } else LatLng(latLng[0], latLng[1])

    }

    fun to(latitude: Double, longitude: Double): LatLng {
        return LatLng(latitude, longitude)
    }
}
